bigtable/tests/filters_integration_test.cc

```diff
@@ -0,0 +1,682 @@
+}
+
+void CheckLatest(bigtable::Client& client, bigtable::Table& table,
+                 std::string const& row_key) {
+  std::vector<bigtable::Cell> created{
+      {row_key, "fam", "c", 0, "v-c-0-0", {}},
+      {row_key, "fam", "c", 1000, "v-c-0-1", {}},
+      {row_key, "fam", "c", 2000, "v-c-0-2", {}},
+      {row_key, "fam0", "c0", 0, "v-c0-0-0", {}},
+      {row_key, "fam0", "c1", 1000, "v-c1-0-1", {}},
+      {row_key, "fam0", "c1", 2000, "v-c1-0-2", {}},
+      {row_key, "fam0", "c1", 3000, "v-c1-0-3", {}},
+  };
+  CreateCells(table, created);
+
+  std::vector<bigtable::Cell> expected{
+      {row_key, "fam", "c", 1000, "v-c-0-1", {}},
+      {row_key, "fam", "c", 2000, "v-c-0-2", {}},
+      {row_key, "fam0", "c0", 0, "v-c0-0-0", {}},
+      {row_key, "fam0", "c1", 2000, "v-c1-0-2", {}},
+      {row_key, "fam0", "c1", 3000, "v-c1-0-3", {}},
+  };
+  auto actual = ReadRow(client, table, row_key, bigtable::Filter::Latest(2));
+  CheckEqual(__func__, expected, actual);
+  std::cout << __func__ << " was successful" << std::endl;
+}
+
+void CheckFamilyRegex(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key) {
+  std::vector<bigtable::Cell> created{
+      {row_key, "fam", "c2", 0, "bar", {}},
+      {row_key, "fam0", "c", 0, "bar", {}},
+      {row_key, "fam1", "c", 0, "bar", {}},
+      {row_key, "fam2", "c", 0, "bar", {}},
+      {row_key, "fam2", "c2", 0, "bar", {}},
+      {row_key, "fam3", "c2", 0, "bar", {}},
+  };
+  CreateCells(table, created);
+
+  std::vector<bigtable::Cell> expected{
+      {row_key, "fam0", "c", 0, "bar", {}},
+      {row_key, "fam2", "c", 0, "bar", {}},
+      {row_key, "fam2", "c2", 0, "bar", {}},
+  };
+  auto actual =
+      ReadRow(client, table, row_key, bigtable::Filter::FamilyRegex("fam[02]"));
+  CheckEqual(__func__, expected, actual);
+  std::cout << __func__ << " was successful" << std::endl;
+}
+
+void CheckColumnRegex(bigtable::Client& client, bigtable::Table& table,
+                      std::string const& row_key) {
+  std::vector<bigtable::Cell> created{
+      {row_key, "fam0", "abc", 0, "bar", {}},
+      {row_key, "fam1", "bcd", 0, "bar", {}},
+      {row_key, "fam2", "abc", 0, "bar", {}},
+      {row_key, "fam3", "def", 0, "bar", {}},
+      {row_key, "fam0", "fgh", 0, "bar", {}},
+      {row_key, "fam1", "hij", 0, "bar", {}},
+  };
+  CreateCells(table, created);
+
+  std::vector<bigtable::Cell> expected{
+      {row_key, "fam0", "abc", 0, "bar", {}},
+      {row_key, "fam2", "abc", 0, "bar", {}},
+      {row_key, "fam0", "fgh", 0, "bar", {}},
+      {row_key, "fam1", "hij", 0, "bar", {}},
+  };
+  auto actual = ReadRow(client, table, row_key,
+                        bigtable::Filter::ColumnRegex("(abc|.*h.*)"));
+  CheckEqual(__func__, expected, actual);
+  std::cout << __func__ << " was successful" << std::endl;
+}
+
+void CheckColumnRange(bigtable::Client& client, bigtable::Table& table,
```

Comment:
I'm not suggesting you rewrite everything written here... but I am wondering, what if this file were written with GoogleTest? It is an integration test, after all, and there's a concept of [one-time global setup][global-setup] (the creation of column families and loading the initial data), and we can even add the [global teardown][global-teardown] (to delete the table or column families), so that it can also be used on a live Bigtable instance and cleanup after itself.

If you agree, consider adding a TODO to investigate use of GoogleTest for this integration test in the future. If not, I'd be curious to hear why think it would be a bad idea in this use case.

[global-setup]: https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#global-set-up-and-tear-down
[global-teardown]: https://github.com/google/googletest/blob/master/googletest/docs/AdvancedGuide.md#global-set-up-and-tear-down

Reply:
Added the bug and TODO.